tests/tests/eth_provider.rs

```diff
@@ -805,6 +807,170 @@ async fn test_send_raw_transaction_wrong_signature(#[future] katana: Katana, _se
+
+    // Create the second transaction request with a higher value
+    let request = TransactionRequest {
+        from: Some(eoa_address),
+        to: Some(TxKind::Call(Address::ZERO)),
+        gas: Some(21000),
+        gas_price: Some(10),
+        value: Some(U256::from(1_000_000)),
+        ..Default::default()
+    };
+
+    // Initialize state override with the EOA address having a lower balance than the required value
+    let mut state_override: HashMap<Address, AccountOverride> = HashMap::new();
+    state_override
+        .insert(eoa_address, AccountOverride { balance: Some(U256::from(1_000_000_000)), ..Default::default() });
+
+    // Attempt to call and handle the result
+    // Should succeed as the EOA balance is higher than the required value
+    let _ = eth_provider
+        .call(request, None, Some(state_override.clone()), None)
+        .await
+        .expect("Failed to call for a simple transfer");
+}
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_call_with_state_override_balance_failure(#[future] katana: Katana, _setup: ()) {
+    // Obtain an Ethereum provider instance from the Katana instance
+    let eth_provider = katana.eth_provider();
+
+    // Get the EOA (Externally Owned Account) address from Katana
+    let eoa_address = katana.eoa().evm_address().expect("Failed to get eoa address");
+
+    // Create the second transaction request with a higher value
+    let request = TransactionRequest {
+        from: Some(eoa_address),
+        to: Some(TxKind::Call(Address::ZERO)),
+        gas: Some(21000),
+        gas_price: Some(10),
+        value: Some(U256::from(1_000_000_001)),
+        ..Default::default()
+    };
+
+    // Initialize state override with the EOA address having a lower balance than the required value
+    let mut state_override: HashMap<Address, AccountOverride> = HashMap::new();
+    state_override
+        .insert(eoa_address, AccountOverride { balance: Some(U256::from(1_000_000_000)), ..Default::default() });
+
+    // Attempt to call and handle the result
+    // Should fail as the EOA balance is lower than the required value
+    match eth_provider.call(request, None, Some(state_override.clone()), None).await {
+        // If call succeeds, panic as an error was expected
+        Ok(_) => panic!("Expected an error but got a success response"),
+        // If call fails, check if the error is due to insufficient funds
+        Err(e) => match e {
+            EthApiError::Transaction(TransactionError::Call(err)) => {
+                assert_eq!(
+                    err.to_string(),
+                    "transaction validation error: lack of funds (1000000000) for max fee (1000210001)",
+                );
+            }
+            // If a different error occurs, panic
+            _ => panic!("Expected a transaction error but got a different error: {e:?}"),
+        },
+    }
```

Comment:
can be replaced by something shorter no?
```rust
let res = eth_provider.call(request, None, Some(state_override.clone()), None).await;
let err = res.unwrap_err().to_string();
assert_eq!(err, "transaction validation error: lack of funds (1000000000) for max fee (1000210001)");
```

Reply:
If you agree as it is a test I would like to leave these different messages like that, even if the test fails, I have an indication of what is happening in the CI (for example another error than the expected one or the fact that I expected a failure but I got a success). I think it can help us understand an error more quickly if one day we have a problem in the test. wdyt?

Reply:
> (for example another error than the expected one or the fact that I expected a failure but I got a success)

I agree but wouldn't `unwrap_err` and the `assert_eq` give you both of this?

Reply:
Yes in theory should be, convinced, let us simplify and if we have issues wit debugging we can revert in the future :)